Lib/Inc/adc_data.h

```diff
@@ -13,21 +13,27 @@
 #include "adc.h"
 #include "log.h"
 
-class CAdc
+/**
+ *  @brief Number of active ADC Channels,
+ *   change this according to hardware
+ */
+#define ADC_CHANNELS 14
```

Comment:
What you could do is allocate 20 slots here, but allow for ctor to have less channels used. It'll be wasted memory, but only very little. To be considered.